packages/lab/src/Flow/Flow.spec.ts

```diff
@@ -0,0 +1,367 @@
+import { test, expect, type Locator, type Page } from "@playwright/test";
+
+let flowCanvasLocator: Locator;
+let referencePage: Page;
+
+async function selectNode(
+  nodeGroup: string,
+  nodeName: string,
+  withExpand: boolean
+) {
+  const baseAssetNodeLocator = referencePage
+    .getByRole("listitem")
+    .filter({ hasText: nodeGroup });
+  if (withExpand) {
+    await baseAssetNodeLocator
+      .getByRole("button", { name: "Expand Group" })
+      .click();
+  }
+  return baseAssetNodeLocator.getByRole("button", { name: nodeName });
+}
+
+async function dragToPosition1(node: Locator) {
+  await node.dragTo(flowCanvasLocator, { targetPosition: { x: 120, y: 50 } });
+}
+
+async function dragToPosition2(node: Locator) {
+  await node.dragTo(flowCanvasLocator, { targetPosition: { x: 130, y: 400 } });
+}
+
+async function dragToPosition3(node: Locator) {
+  await node.dragTo(flowCanvasLocator, { targetPosition: { x: 620, y: 80 } });
+}
+
+async function connectNodes(
+  sourceNode: string,
+  sourceConnection: string,
+  destinyNode: string,
+  destinyConnection
+) {
+  const source = flowCanvasLocator
+    .getByRole("button", { name: sourceNode })
+    .locator(".react-flow__handle-right")
+    .filter({
+      has: referencePage
+        .locator("../div")
+        .getByText(sourceConnection, { exact: true }),
+    });
+  const destiny = flowCanvasLocator
+    .getByRole("button", { name: destinyNode })
+    .locator(".react-flow__handle-left")
+    .filter({
+      has: referencePage
+        .locator("../div")
+        .getByText(destinyConnection, { exact: true }),
+    });
+  await source.dragTo(destiny, { force: true });
+}
+
+async function deleteConnection(connectionName: string) {
+  await referencePage.getByRole("button", { name: connectionName }).click();
+  await referencePage.keyboard.press("Backspace");
+}
+
+async function moveUp(renderedNode: Locator, verticalMovement: number) {
+  const renderedNodeBox = await renderedNode.boundingBox();
+  const initialPointX = renderedNodeBox!.x + 30;
+  const initialPointY = renderedNodeBox!.y + 30;
+  await referencePage.mouse.move(initialPointX, initialPointY);
+  await referencePage.mouse.down();
+  await referencePage.mouse.move(
+    initialPointX,
+    initialPointY - verticalMovement
+  );
+  await referencePage.mouse.up();
+}
+
+test.beforeEach(async ({ page }) => {
+  referencePage = page;
+  flowCanvasLocator = page.locator("id=hvFlow:r1:");
```

Comment:
This selector is brittle, as this is a generated `id` and there's no guarantees it won't change.
We could use the `HvFlow-root` class or the `react-flow` class (in the child element) as these are stable